src/main/java/com/github/tomakehurst/wiremock/common/ResourceUtil.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2023 Thomas Akehurst
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.tomakehurst.wiremock.common;
+
+import static com.github.tomakehurst.wiremock.common.Exceptions.throwUnchecked;
+import static com.github.tomakehurst.wiremock.common.ParameterUtils.*;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class ResourceUtil {
+
+  private ResourceUtil() {}
+
+  public static <T> ClassLoader getLoader(Class<T> className) {
+    return getFirstNonNull(
+        Thread.currentThread().getContextClassLoader(), className.getClassLoader());
```

Comment:
I am not sure it is a right priority order, and most likely not the one expected in `getResource()` where you would start from the class and then, maybe?, try other classloaders

Reply:
Hello.
I completely repeated com.google.common.io.Resources

```
public static URL getResource(String resourceName) {
    ClassLoader loader =
        MoreObjects.firstNonNull(
            Thread.currentThread().getContextClassLoader(), Resources.class.getClassLoader());
...
```

I can change and test if needed.